src/Microsoft.VisualStudio.ProjectSystem.Managed/PooledObjects/StringTableLifetimeService.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.Buffers.PooledObjects
+{
+    /// <summary>
```

Comment:
Can you explain what this is trying to acheive? Are you seeing that we're not sharing a lot of strings between configurations?

Reply:
Also, this object lives per implicitly active configuration.

Reply:
Not sure if you're asking a general or specific question here. The general observation is that a lot of the space in Dependency objects is wasted because Dependency IDs are so long and generated on the fly rather than centralized (so we don't have much of any sharing). I haven't reviewed this PR yet in detail, it might be overkill since it would be a huge improvement just to intern within a project (I think in some cases I was looking at, there were 8 copies of a particular Dependency ID per project.)

Reply:
I was referring to this class exactly. I understand what we're trying to solve with the interning.